sale_commission_settlement_payment_date/tests/test_commission_payment_date.py

```diff
@@ -0,0 +1,167 @@
+        wizard.with_context(
+            {
+                "active_model": "sale.order",
+                "active_ids": [sale_order.id],
+                "active_id": sale_order.id,
+            }
+        ).create_invoices()
+        invoice = sale_order.invoice_ids - old_invoices
+        if date:
+            invoice.invoice_date = date
+            invoice.date = date
+        # We need to use flush() in order to execute commission_amount
+        invoice.flush()
+        # invoice.payment_state = "paid"
+        return invoice
+
+    def _settle_agent(self, agent=None, period=None, date=None):
+        vals = {
+            "date_to": (
+                fields.Datetime.from_string(fields.Datetime.now())
+                + dateutil.relativedelta.relativedelta(months=period)
+            )
+            if period
+            else date,
+            "date_payment_to": datetime.now(),
+        }
+        if agent:
+            vals["agent_ids"] = [(4, agent.id)]
+        wizard = self.make_settle_model.create(vals)
+        wizard.action_settle()
+
+    def register_payment(self, invoice):
+        payment_model = self.env["account.payment.register"]
+        invoice.action_post()
+        return (
+            payment_model.with_context(
+                {"active_model": "account.move", "active_ids": invoice.ids}
+            )
+            .create(
+                {
+                    "payment_date": datetime.now() - timedelta(days=2),
+                    "amount": invoice.amount_total,
+                    "journal_id": self.journal.id,
+                    "payment_method_id": self.env.ref(
+                        "account.account_payment_method_manual_out"
+                    ).id,
+                }
+            )
+            .action_create_payments()
+        )
+
+    def test_commission_single_invoice(self):
+        sale_order = self._create_sale_order(
+            self.agent_monthly, self.commission_net_paid
+        )
+        sale_order.action_confirm()
+        self._invoice_sale_order(sale_order)
+        date = fields.Date.today()
+        sale_order.invoice_ids.write(
+            {
+                "invoice_date": date + relativedelta(months=-1),
+                "date": date + relativedelta(months=-1),
+            }
+        )
+        sale_order.invoice_ids.action_post()
+        sale_order = self._create_sale_order(
+            self.agent_monthly, self.commission_net_paid
+        )
+        sale_order.action_confirm()
+        inv = self._invoice_sale_order(sale_order)
+        self.register_payment(inv)
+        self._settle_agent(self.agent_monthly, 1)
+        settlements = self.env["sale.commission.settlement"].search(
+            [
+                (
+                    "agent_id",
+                    "=",
+                    self.agent_monthly.id,
+                ),
+                ("state", "=", "settled"),
+            ]
+        )
+        self.assertEqual(1, len(settlements))
```

Comment:
I'm not entirely sure, so please correct me if I'm wrong, but it seems to me that this test only creates a single payment and checks that it's included in the wizard.

You could improve the test by creating two payments (or two sales orders with one payment each, if it's easier), one in the past (as the one you've already created) and one in the future, and making sure that when you call the wizard, only the one before `date_payment_to` is actually included in the wizard and the other is not.

As it is, if I understand correctly, your module could do nothing and this test would still pass.